File: TP1/src/binaire.c
#include <stdio.h>
#include "binaire.h"

/*
* binaire.c
*
* Objectif :
* Convertir et afficher un nombre entier en format binaire
*
* Auteur :
* Mohamed AIT EL HADJ
*
* Date :
* 16/09/2024
*/

void binaire() {
    int num, i;
    int binaire[32];

    printf("Entrez un nombre entier: ");
    scanf("%d", &num);

    for (i = 31; i >= 0; i--) {
        binaire[i] = num % 2;
        num /= 2;
    }

    printf("Representation binaire: ");
    for (i = 0; i < 32; i++) {
        printf("%d", binaire[i]);
    }
    printf("\n");
}

File: TP1/src/binaire.py
"""
binaire.py

Objectif :
Convertir et afficher un nombre entier en format binaire

Auteur :
Mohamed AIT EL HADJ

Date :
16/09/2024
"""

num = int(input("Entrez un nombre entier: "))
binaire = []

while num > 0:
    binaire.append(num % 2)
    num //= 2

binaire.reverse()
print("Représentation binaire:", ''.join(map(str, binaire)))
